lib/systemd.c

```diff
@@ -127,3 +104,55 @@ void systemd_send_status(const char *status)
 	snprintf(buffer, sizeof(buffer), "STATUS=%s", status);
 	systemd_send_information(buffer);
 }
+
+static intmax_t getenv_int(const char *varname, intmax_t dflt)
+{
+	char *val, *err;
+	intmax_t intval;
+
+	val = getenv(varname);
+	if (!val)
+		return dflt;
+
+	intval = strtoimax(val, &err, 0);
+	if (*err || !*val)
+		return dflt;
+	return intval;
+}
+
+void systemd_init_env(void)
+{
+	char *tmp;
+	uintmax_t dev, ino;
+	int len;
+	struct stat st;
+
+	notify_socket = getenv("NOTIFY_SOCKET");
+
+	/* no point in setting up watchdog w/o notify socket */
+	if (notify_socket) {
+		watchdog_pid = getenv_int("WATCHDOG_PID", -1);
+		if (watchdog_pid <= 0)
+			watchdog_pid = -1;
+		watchdog_usec = getenv_int("WATCHDOG_USEC", 0);
+		if (watchdog_usec < 0)
+			watchdog_usec = 0;
```

Comment:
```c
	thread_add_timer_msec(systemd_master, systemd_send_watchdog, NULL,
			      watchdog_usec / 1000, NULL);
```
What if `watchdog_usec` is 0? Do we peg a core with this timer?

Reply:
It doesn't get started for 0 due to line 96:

```c
	if (watchdog_usec != 0)
		systemd_send_watchdog(NULL);
```

But you're still right, for values 1-999 since it's divided by 1000 in `systemd_send_watchdog` it'll still end up 0.  I'll go fix that.